backends/arm/test/ops/test_full.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+#
+# Tests the full op which creates a tensor of a given shape filled with a given value.
+# The shape and value are set at compile time, i.e. can't be set by a tensor input.
+#
+
+import unittest
+from typing import Tuple
+
+import torch
+from executorch.backends.arm.test import common
+from executorch.backends.arm.test.tester.arm_tester import ArmTester
+from parameterized import parameterized
+
+
+class TestFull(unittest.TestCase):
+    class Full(torch.nn.Module):
+        # A single full op
+        def forward(self):
+            return torch.full((3, 3), 4.5)
+
+    class AddConstFull(torch.nn.Module):
+        # Input + a full with constant value.
+        def forward(self, x: torch.Tensor):
+            return torch.full((2, 2, 3, 3), 4.5, dtype=torch.float32) + x
+
+    class AddVariableFull(torch.nn.Module):
+        sizes = [
+            (5),
+            (5, 5),
+            (5, 5, 5),
+            (1, 5, 5, 5),
+        ]
+        test_parameters = [((torch.randn(n) * 10 - 5, 3.2),) for n in sizes]
+
+        def forward(self, x: torch.Tensor, y):
+            # Input + a full with the shape from the input and a given value 'y'.
+            return x + torch.full(x.shape, y)
+
+    def _test_full_tosa_MI_pipeline(
+        self,
+        module: torch.nn.Module,
+        example_data: torch.Tensor,
+        test_data: torch.Tensor = None,
+    ):
+        if test_data is None:
+            test_data = example_data
+        (
+            ArmTester(
+                module,
+                example_inputs=example_data,
+                compile_spec=common.get_tosa_compile_spec(),
+            )
+            .export()
+            .check_count({"torch.ops.aten.full.default": 1})
+            .to_edge()
+            .partition()
+            .check_not(["executorch_exir_dialects_edge__ops_aten_full_default"])
+            .check_count({"torch.ops.higher_order.executorch_call_delegate": 1})
+            .to_executorch()
+            .run_method_and_compare_outputs(inputs=test_data)
+        )
+
+    def _test_full_tosa_BI_pipeline(
+        self,
+        module: torch.nn.Module,
+        test_data: Tuple[torch.Tensor],
+        permute_memory_to_nhwc: bool,
+    ):
+        (
+            ArmTester(
+                module,
+                example_inputs=test_data,
+                compile_spec=common.get_tosa_compile_spec(
+                    permute_memory_to_nhwc=permute_memory_to_nhwc
+                ),
+            )
+            .quantize()
```

Comment:
I am open to suggestions if you have ideas around how to remove dedup across MI and BI where just this call is different.

Reply:
We have discussed this internally, and while I agree that seems redundant, we have decided that we prefer to be able to have different pipelines for easily customizing them under development/ in the future

Reply:
after writing this I was thinking something like,

`def test_foo(quantize:bool = True): 
    tester = Tester(..)...
    if quantize:
		tester = tester.quantize()
`

But I guess if you are doing checks etc around quantization then it might be more complicated. Ok let me resolve this.